### PR TITLE
Add Telink docker image

### PR DESCRIPTION
**Pull request by rikorsev**
**Problem**

To integrate and validate Telink port of CHIP, docker images are required to build the examples in the Github workflow.
Like other docker images used by this repository, they should be available in the connectedhomeip docker hub.

**Summary of Changes**

Telink docker image build files are located integrations/docker/images/chip-build-telink.

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

File: integrations/docker/images/chip-build-telink/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+RUN apt-get update
+RUN apt-get install -y libc6:i386 libncurses5:i386 libstdc++6:i386
+RUN mkdir ${TELINK_DIR}
+RUN cd ${TELINK_DIR} && wget http://wiki.telink-semi.cn/tools_and_sdk/Tools/IDE/telink_riscv_linux_toolchain.zip
+RUN unzip ${TELINK_DIR}/telink_riscv_linux_toolchain.zip -d ${TELINK_DIR}
+RUN rm ${TELINK_DIR}/telink_riscv_linux_toolchain.zip
+
+# Setup Zephyr dependencies
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt install -y git
```
[the remark this answers is not in the document]
Comment (woody-apple) [reply integrations/docker/images/chip-build-telink/Dockerfile:18]: Note: the reason for this is if you chain commands like this, it's a single docker image snapshot, vs each RUN in this one will create a snapshot for each - thus making the image size larger.

**Review by woody-apple — CHANGES_REQUESTED** [review]
Please collapse RUN statements like other docker images into one :)

File: integrations/docker/images/chip-build-telink/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (rikorsev) [reply integrations/docker/images/chip-build-telink/Dockerfile:18]: Hi @andy31415 and @woody-apple 

Thanks for feedback! It will be fixed.

**Comment by andy31415** [comment]
@woody-apple  for review?

@rikorsev - please add a 'how did you test this' section in the PR summary. In this case I believe building the docker image locally and eventually connecting to it via `run -it` and some path validation could be sufficient.

**Review by woody-apple — APPROVED** [review]

File: integrations/docker/images/chip-build-telink/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    && pip3 install --user -U west \
+    && pip3 install pyelftools \
+    && west init ${ZEPHYR_PROJECT_DIR} \
+    && cd ${ZEPHYR_PROJECT_DIR} && west update && west zephyr-export \
+    && python3 -m pip install -U pip \
+    && pip3 install --user -r ${ZEPHYR_PROJECT_DIR}/zephyr/scripts/requirements.txt
+
+# Workaround below is require due to Telink platform
+# is not included into official Zephyr repo yet.
+# Below step will be removed in future.
```
Comment (mspang) [inline integrations/docker/images/chip-build-telink/Dockerfile:41]: ETA?

**Pull request merged**